tests/test_remapper.py

```diff
@@ -1,2 +1,124 @@
-def test_remapper():
-    assert 2 == 2
+import dask.array
+import numpy as np
+import pytest
+import sparse
+import xarray as xr
+
+from xgriddedaxis import Remapper
+from xgriddedaxis.remapper import _INCOMING_KEY, _OUTGOING_KEY
+
+from .utils import create_dataset, generate_time_and_bounds
+
+
+@pytest.fixture(scope='module')
+def incoming():
+    n = 13
+    bounds = np.round(np.logspace(2.0, 3.5, num=n), decimals=0)
+    fractions = np.round(np.random.random(n - 1), decimals=1)
+    incoming = generate_time_and_bounds(bounds, fractions)
+    return incoming
+
+
+@pytest.fixture(scope='module')
+def outgoing():
+    n = 25
+    bounds = np.round(np.logspace(2.0, 3.6, num=n), decimals=0)
+    fractions = np.round(np.random.random(n - 1), decimals=1)
+    outgoing = generate_time_and_bounds(bounds, fractions)
+    return outgoing
+
+
+@pytest.fixture(scope='module')
+def dataset(incoming, outgoing):
+    ds = create_dataset(incoming['time'], incoming['time_bounds'])
+    return ds
+
+
+def test_remapper_init(incoming, outgoing):
+    remapper = Remapper(
+        incoming_axis=incoming,
+        outgoing_axis=outgoing,
+        axis_name='time',
+        boundary_variable='time_bounds',
+    )
+    assert isinstance(remapper.weights.data, sparse._coo.core.COO)
+    assert remapper.weights.shape == (
+        remapper.outgoing_axis[remapper.axis_name].size,
+        remapper.incoming_axis[remapper.axis_name].size,
+    )
+    assert set(remapper.weights.dims) == set([_OUTGOING_KEY, _INCOMING_KEY])
```

Comment:
This does not check the order of the `dims`, but the order matters, right?  Or does it not because it's `xarray`?

Reply:
I am setting the coordinates in `construct_coverage_matrix()`, and relying on xarray to take of the alignment of indices and dimension sizes. 

```python
weights = xr.DataArray(data=wgts, dims=['outgoing', 'incoming'], coords=coords)

```